services/bettersocial/fcmToken/sendChatNotification.js

```diff
@@ -0,0 +1,20 @@
+const {messaging} = require('firebase-admin');
+const FcmTokenFunction = require('../../../databases/functions/fcmToken');
+const {FcmToken, User} = require('../../../databases/models');
+const UsersFunction = require('../../../databases/functions/users');
+
+const sendChatNotification = async (userTargetId, payload) => {
+  const signedUser = await UsersFunction.findSignedUserId(User, userTargetId);
+  const userTargetToken = await FcmTokenFunction.findTokenByUserId(FcmToken, signedUser);
+  if (userTargetToken) {
+    try {
+      messaging()
+        .sendToDevice(userTargetToken?.token, payload)
+        .then(() => {});
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The `sendToDevice` function is a promise that is not being handled properly. The `.then(() => {})` is essentially swallowing any errors that might occur during the execution of this promise. It would be better to handle this promise inside the try-catch block and remove the `.then(() => {})`.
```diff
-      messaging()
-        .sendToDevice(userTargetToken?.token, payload)
-        .then(() => {});
+      await messaging().sendToDevice(userTargetToken?.token, payload);
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->